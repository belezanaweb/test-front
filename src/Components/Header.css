.App-header {
  background-color: #FFFFFF;
  min-height: 6vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 1vmin);
  letter-spacing: calc(-1px + 0.1vmin);
  font-weight: 600;
  color: hsl(0, 0%, 100%);
  box-shadow: 0 1px #FFFFFF inset, 0 0 3px hsla(0, 15%, 12%, 0.4);
  padding: 2px;
}

.App-header .App-navbar {
  color: hsl(0, 0%, 82.5%);
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  width: 100%;
}

.App-header .App-navbar ul {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  list-style: none;
  width: 88vw;
  padding: 0;
}

.App-header .App-navbar ul li {
  display: inline-block;
  text-transform: uppercase;
}

.App-header .App-navbar ul li a {
  text-decoration: none;
  color: rgb(210, 210, 210);
}

.selected a {
  color: rgb(255, 120, 0) !important;
}
