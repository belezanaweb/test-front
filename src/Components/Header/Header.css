.Header {
  display: flex;
  align-items: center;
  height: 40px;
  margin: 32px 13px 0 14px;
  border-radius: 3px;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0, 0, 29, 0.22);
  justify-content: center;
  z-index: 1;
}

.Header ul {
  list-style: none;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
}

.Header li {
  cursor: pointer;
  height: 16px;
  color: #ccc;
  font-family: 'Helvetica Neue';
  font-size: 13px;
  font-weight: 700;
  text-align: center;
  margin: 0 15px;
}

.Header li:hover {
  color: #ff7800;
}
