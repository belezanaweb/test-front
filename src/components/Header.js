import { Grid, makeStyles, Typography } from '@material-ui/core'
import React from 'react'

const useStyles = makeStyles({
  root: {
    background: '#FFF',
    height: '40px'
  }
})

function Header({ select }) {
  const classes = useStyles()
  const selected = select

  return (
    <Grid
      container
      direction="row"
      justifyContent="space-evenly"
      alignItems="center"
      className={classes.root}
    >
      <Grid item>
        <Typography style={{ color: selected === 1 ? '#FF7800' : '#CCC', fontWeight: 'bold' }}>
          SACOLA
        </Typography>
      </Grid>
      <Grid item>
        <Typography style={{ color: selected === 2 ? '#FF7800' : '#CCC', fontWeight: 'bold' }}>
          PAGAMENTO
        </Typography>
      </Grid>
      <Grid item>
        <Typography style={{ color: selected === 3 ? '#FF7800' : '#CCC', fontWeight: 'bold' }}>
          CONFIRMAÇÃO
        </Typography>
      </Grid>
    </Grid>
  )
}
export default Header
