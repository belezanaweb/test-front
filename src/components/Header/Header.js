import { useHistory } from 'react-router'

const Header = (props) => {
  const history = useHistory()

  return (
    <div>
      {history.location.pathname === '/cart' ? (
        <p onClick={() => goToCart(history)}>SACOLA</p>
      ) : (
        <p onClick={() => goToCart(history)}>SACOLA</p>
      )}

      {history.location.pathname === '/payment' ? (
        <p onClick={() => goToPayment(history)}>PAGAMENTO</p>
      ) : (
        <p onClick={() => goToPayment(history)}>PAGAMENTO</p>
      )}

      {history.location.pathname === '/confirmation' ? (
        <p onClick={() => goToConfirmation(history)}>CONFIRMAÇÃO</p>
      ) : (
        <p onClick={() => goToConfirmation(history)}>CONFIRMAÇÃO</p>
      )}
    </div>
  )
}

export default Header
