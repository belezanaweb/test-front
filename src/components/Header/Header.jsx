import React from 'react'
import * as CSS from './style'
import { useLocation } from 'react-router-dom'

export default function Header() {
  const location = useLocation()
  const path = location.pathname || ''

  return (
    <CSS.Header>
      <CSS.Menu>
        <CSS.Item active={path === '' || path === '/' || path.includes('cart')}>Sacola</CSS.Item>
        <CSS.Item active={path.includes('payment')}>Pagamento</CSS.Item>
        <CSS.Item active={path.includes('success')}>Confirmação</CSS.Item>
      </CSS.Menu>
    </CSS.Header>
  )
}
