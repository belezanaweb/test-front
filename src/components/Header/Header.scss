@import "../../variables";

.Header {
  display: flex;
  justify-content: space-between;
  padding: 0.9231rem 2.385rem;
  background-color: $white;
  box-shadow: 1px 1px 5px 0 rgba(0, 0, 29, 0.22);

  > a {
    line-height: 16px;
    font-weight: $bold;
    text-transform: uppercase;
    text-decoration: none;
    color: $gray;
    transition: color $transition;

    &.active {
      color: $primary;
    }
  }
}
