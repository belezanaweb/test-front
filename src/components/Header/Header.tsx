import { PropsWithChildren } from 'react'

function Header({ children }: PropsWithChildren) {
  return <header className="bg-white shadow-md justify-center px-5">{children}</header>
}

export default Header
