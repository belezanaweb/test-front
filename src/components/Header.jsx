import React from 'react'
import { Link } from 'react-router-dom'
import '../styles/Header.css'

export default class Header extends React.Component {
  state = {
    isCart: false,
    isPayment: false,
    isSubmit: false
  }
  componentDidMount() {
    this.setColor()
  }
  setColor = () => {
    if (window.location.pathname === '/') {
      this.setState({
        isCart: true,
        isPayment: false,
        isSubmit: false
      })
    }

    if (window.location.pathname === '/payment') {
      this.setState({
        isCart: false,
        isPayment: true,
        isSubmit: false
      })
    }

    if (window.location.pathname === '/submit') {
      this.setState({
        isCart: false,
        isPayment: false,
        isSubmit: true
      })
    }
  }
  render() {
    const { isCart, isSubmit, isPayment } = this.state
    return (
      <div className="header-container">
        <Link to="/" className={`header-link ${isCart ? 'focus' : ''}`}>
          Sacola
        </Link>
        <Link to="/payment" className={`header-link ${isPayment ? 'focus' : ''}`}>
          Pagamento
        </Link>
        <span className={`header-link ${isSubmit ? 'focus' : ''}`}>Confirmação</span>
      </div>
    )
  }
}
