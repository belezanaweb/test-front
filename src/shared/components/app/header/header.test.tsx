import React from 'react';
import { render, screen } from '../../../../../config/test-utils';

import Header from './index';

describe('<Header />', () => {
  it('should render component with <header> tag', () => {
    render(<Header>Header</Header>, {});
    expect(screen.findByRole('header'));
  })

  it('should render component with default props and match snapshot', () => {
    const { container } = render(<Header>Header</Header>, {});
    expect(container.firstChild).toMatchSnapshot();
  })
})
