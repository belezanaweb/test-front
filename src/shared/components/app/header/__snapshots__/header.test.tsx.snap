// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render component with default props and match snapshot 1`] = `
.c0 {
  background: #fff;
  box-shadow: 1px 1px 5px 0 #00001d38;
  padding: 1rem;
}

<header
  class="c0"
  role="banner"
>
  Header
</header>
`;
